core/src/main/java/org/springframework/security/scheduling/DelegatingSecurityContextTaskScheduler.java

```diff
@@ -32,45 +36,66 @@
 
-	private final TaskScheduler taskScheduler;
+	private final TaskScheduler delegate;
+	private final SecurityContext securityContext;
 
 	/**
-	 * Creates a new {@link DelegatingSecurityContextTaskScheduler}
-	 * @param taskScheduler the {@link TaskScheduler}
+	 * Creates a new {@link DelegatingSecurityContextTaskScheduler} that uses the
+	 * specified {@link SecurityContext}.
+	 * @param delegateTaskScheduler the {@link TaskScheduler} to delegate to. Cannot be
+	 * null.
+	 * @param securityContext the {@link SecurityContext} to use for each
+	 * {@link DelegatingSecurityContextRunnable} or null to default to the current
+	 * {@link SecurityContext}
 	 */
-	public DelegatingSecurityContextTaskScheduler(TaskScheduler taskScheduler) {
-		Assert.notNull(taskScheduler, "Task scheduler must not be null");
-		this.taskScheduler = taskScheduler;
+	public DelegatingSecurityContextTaskScheduler(TaskScheduler delegateTaskScheduler,
+			SecurityContext securityContext) {
+		Assert.notNull(delegateTaskScheduler, "delegateTaskScheduler cannot be null");
+		this.delegate = delegateTaskScheduler;
+		this.securityContext = securityContext;
+	}
+
+	/**
+	 * Creates a new {@link DelegatingSecurityContextTaskScheduler} that uses the current
+	 * {@link SecurityContext} from the {@link SecurityContextHolder}.
+	 * @param delegate the {@link TaskExecutor} to delegate to. Cannot be null.
+	 */
+	public DelegatingSecurityContextTaskScheduler(TaskScheduler delegate) {
+		this(delegate, null);
 	}
 
 	@Override
 	public ScheduledFuture<?> schedule(Runnable task, Trigger trigger) {
-		return this.taskScheduler.schedule(task, trigger);
+		return this.delegate.schedule(wrap(task), trigger);
 	}
 
 	@Override
 	public ScheduledFuture<?> schedule(Runnable task, Date startTime) {
-		return this.taskScheduler.schedule(task, startTime);
+		return this.delegate.schedule(wrap(task), startTime);
 	}
 
 	@Override
 	public ScheduledFuture<?> scheduleAtFixedRate(Runnable task, Date startTime, long period) {
-		return this.taskScheduler.scheduleAtFixedRate(task, startTime, period);
+		return this.delegate.scheduleAtFixedRate(wrap(task), startTime, period);
 	}
 
 	@Override
 	public ScheduledFuture<?> scheduleAtFixedRate(Runnable task, long period) {
-		return this.taskScheduler.scheduleAtFixedRate(task, period);
+		return this.delegate.scheduleAtFixedRate(wrap(task), period);
 	}
 
 	@Override
 	public ScheduledFuture<?> scheduleWithFixedDelay(Runnable task, Date startTime, long delay) {
-		return this.taskScheduler.scheduleWithFixedDelay(task, startTime, delay);
+		return this.delegate.scheduleWithFixedDelay(wrap(task), startTime, delay);
 	}
 
 	@Override
 	public ScheduledFuture<?> scheduleWithFixedDelay(Runnable task, long delay) {
-		return this.taskScheduler.scheduleWithFixedDelay(task, delay);
+		return this.delegate.scheduleWithFixedDelay(wrap(task), delay);
+	}
+
+	protected final Runnable wrap(Runnable delegate) {
```

Comment:
Let's please leave this private for now until there is a clear use case for making it protected. Users who extend this class can easily create their own wrap method.

Reply:
`wrap` is now private